blues/moves.py

```diff
@@ -0,0 +1,256 @@
+        #multiply lig coordinates by rot matrix and add back COM translation from origin
+        rot_move = np.dot(reduced_pos, rand_rotation_matrix) * positions.unit + self.center_of_mass
+
+        # Update ligand positions in nc_sim
+        for index, atomidx in enumerate(self.atom_indices):
+            positions[atomidx] = rot_move[index]
+        context.setPositions(positions)
+        positions = context.getState(getPositions=True).getPositions(asNumpy=True)
+        self.positions = positions[self.atom_indices]
+        return context
+
+
+class CombinationMove(Move):
+    """Move object that allows Move object moves to be performed sequentially.
+    These moves are performed according to the order in move_list. To ensure
+    detailed balance, the moves have an equal chance to be performed in listed
+    or reverse order.
+    Parameters
+    ----------
+    move_list : list of blues.move.Move-like objects
+    """
+    def __init__(self, move_list):
+        self.move_list = move_list
+
+    def move(self, context):
+        rand = np.random.random()
+        #to maintain detailed balance this executes both
+        #the forward and reverse order moves with equal probability
+        if rand > 0.5:
+            for single_move in self.move_list:
+                 single_move.move(context)
+        else:
+            for single_move in reverse(self.move_list):
+                 single_move.move(context)
+
+
+class MoveEngine(object):
+    """MoveEngine provides perturbation functions for the context during the NCMC
+    simulation.
+    Ex.
+        from blues.ncmc import MoveProposal
+        mover = MoveProposal(model, 'random_rotation', nstepsNC)
+        #Get the dictionary of proposed moves
+        mover.moves
+    """
+    def __init__(self, moves, probabilities=None):
+        """Initialize the MovePropsal object that contains functions to perturb
+        the context in the NCMC simulation.
+        Parameters
+        ----------
+        moves : blues.ncmc.Model object or list of blues.ncmc.Model-like objects
+            Specifies the possible moves to be performed
+
+        probabilities:
+            A list of n probabilities, which correspond to the likelyhood
+            of a Move object from moves being selected to perform its
+            corresponding move.
+            If probabilities=None (the default setting) then uniform
+            probabilities are given to each move
+        """
+
+    #make a list from moves if not a list
+        if isinstance(moves,list):
+            self.moves = moves
+        else:
+            self.moves = [moves]
+        #normalize probabilities
+        if probabilities is None:
+            single_prob = 1. / len(self.moves)
+            self.probs = [single_prob for x in (self.moves)]
+        else:
+            prob_sum = float(sum(probabilities))
+            self.probs = [x/prob_sum for x in probabilities]
+        assert len(self.moves) == len(self.probs)
```

Comment:
Is there a reason for this to be an assertion rather than an Exception? I tend to prefer raising useful exceptions if a condition is not met which might not be met because of user error, and reserving assertions for conditions which should never occur. I think I got this from some sort of "best practices" guide but it resonated with me, because an assertion prints out failure info which is less useful than an exception. So the logic is, if you want to tell a user they've screwed up, use an exception; if you're just checking that the code is not broken, use an assertion.